src/main/java/org/testng/TestNG.java

```diff
@@ -975,19 +975,15 @@ private void initializeConfiguration() {
     // Install the listeners found in the suites
     //
     for (XmlSuite s : m_suites) {
-      for (String listenerName : s.getListeners()) {
-        Class<?> listenerClass = ClassHelper.forName(listenerName);
-
-        // If specified listener does not exist, a TestNGException will be thrown
-        if(listenerClass == null) {
-          throw new TestNGException("Listener " + listenerName
-              + " was not found in project's classpath");
-        }
-
-        Object listener = ClassHelper.newInstance(listenerClass);
-        addListener(listener);
+      // Add the child suite listeners
+      List<XmlSuite> childSuites = s.getChildSuites();
```

Comment:
@email2vimalraj 
This is now going to run the listeners that were originally supposed to be part of a child suite only, to be executed for all suites.

For e.g., Lets consider the following :

1. Parent suite has 3 listeners `(p1, p2, p3)` added via the `listeners` tag.
2. Child suite1 has 2 listeners `(c1, c2)` added via the `listeners` tag.
3. Child suite2 has no listeners attached to it.

The expected behavior IMO should be as below

When Child suite1 is executed 5 listeners should be executed `(p1,p2,p3,c1,c2)` (because `p1,p2,p3` are part of the parent suite and so its global in scope)

When Child suite2 is executed 3 listeners should be executed `(p1,p2,p3)` alone because they are global in nature.

The current changes I believe will end up causing 5 listeners to be executed for Child suite2 as well because its accumulating all the listeners that are defined across the child suites and then they are being executed for all the suites..

Am not very sure if that is desirable ?

@juherr your thoughts ?